rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/shared_functions.hpp

```diff
@@ -229,6 +229,13 @@ wait(const char * implementation_identifier,
           }
         }
         if (fixed) {
+          // Reset the fixed guard conditions to avoid being woken up
+          // immediately next time.
+          DDSGuardCondition * guard = static_cast<DDSGuardCondition *>((*attached_conditions)[i]);
+          retcode = guard->set_trigger_value(DDS_BOOLEAN_FALSE);
+          if (retcode != DDS_RETCODE_OK) {
+            RMW_SET_ERROR_MSG("failed to set trigger value");
```

Comment:
It places the error string in a thread specific storage buffer and the calling code can fetch it when it receives the non-ok return code. So you set the error, return an error code, the calling code notices, and finally fetches the error. It's a common pattern for passing more information along in C error handling.

I'd recommend using `fprintf(stderr, "%s\n", ...);`.